File: src/main/kotlin/acc/richclient/dialogs/InitShowDialog.java
package acc.richclient.dialogs;

import acc.business.Facade;
import acc.richclient.MainWindow;
import acc.util.AccException;
import acc.util.Messages;
import acc.richclient.views.PaneTabs;
import javafx.scene.Node;
import javafx.scene.control.Label;
import javafx.scene.layout.GridPane;

public class InitShowDialog extends AbstractDialog {

    private AccountCB accCB;

    public InitShowDialog() throws AccException {
        super(Messages.Zobraz_pocatecni_stavy.cm());
    }

    @Override
    protected Node createContent() {
        GridPane gp = genGP();
        int row = 0;
        gp.add(new Label(Messages.pro_ucet.cm() + DEL), 0, row);
        gp.add(accCB = new AccountCB(), 1, row);
        try {
            accCB.getItems().addAll(Facade.INSTANCE.getBalanceAccounts());
        } catch (AccException ex) {
            MainWindow.showException(ex);
        }
        return gp;
    }

    @Override
    public void ok() throws AccException {
/*        PaneTabs.Companion.addTab(Messages.Pocatecni_stavy.cm(),
                new InitsPane(accCB.getOptAccount()));*/
    }

    @Override
    public void validate() {
    }
}

File: src/main/kotlin/acc/richclient/dialogs/InitShowDialog.kt
package acc.richclient.dialogs

import acc.business.Facade
import acc.model.AnalAcc
import acc.richclient.panes.InitFragment
import acc.richclient.views.PaneTabs
import acc.util.Messages
import javafx.beans.property.SimpleObjectProperty
import tornadofx.*

class InitShowDialog : Fragment() {

    val acc = SimpleObjectProperty<AnalAcc>()

    override val root =
            form {
                title = Messages.Zobraz_pocatecni_stavy.cm()
                fieldset {
                    field(Messages.Pro_ucet.cm() + AbstractDialog.DEL) {
                        combobox(acc, Facade.allAccounts)
                    }

                }
                buttonbar {
                    button(Messages.Potvrd.cm()) {
                        action {
                            PaneTabs.addTab(Messages.Pocatecni_stavy.cm(),
                                    find<InitFragment>(mapOf("acc" to acc)).root)
                            close()
                        }

                    }
                    button(Messages.Zrus.cm()) {
                        action {
                            close()
                        }
                    }
                }
            }
}